Tests/AEPOptimizeTests/UnitTests/OptimizePublicAPITests.swift

```diff
@@ -563,6 +560,455 @@ class OptimizePublicAPITests: XCTestCase {
         // verify
         wait(for: [expectation], timeout: 1)
     }
+    
+    // MARK: - Mobile Surface Support Tests
+    
+    func testUpdatePropositionsForSurfacePaths_validSurface() {
+        // setup
+        let expectation = XCTestExpectation(description: "updatePropositions should dispatch an event with expected data.")
+        expectation.assertForOverFulfill = true
+
+        let testEventData: [String: Any] = [
+            "requesttype": "updatepropositions",
+            "surfaces": [
+                "myView#htmlElement"
+            ]
+        ]
+        let testEvent = Event(name: "Optimize Update Propositions Request",
+                              type: "com.adobe.eventType.optimize",
+                              source: "com.adobe.eventSource.requestContent",
+                              data: testEventData)
+
+        // test
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(type: testEvent.type,
+                                                                                    source: testEvent.source) { event in
+            XCTAssertEqual(testEvent.name, event.name)
+            XCTAssertEqual("updatepropositions", event.data?["requesttype"] as? String)
+            guard let surfaces = event.data?["surfaces"] as? [String], !surfaces.isEmpty else {
+                XCTFail("Surface string array should be valid.")
+                return
+            }
+            XCTAssertEqual(1, surfaces.count)
+            XCTAssertEqual("myView#htmlElement", surfaces[0])
+
+            XCTAssertNil(event.data?["xdm"])
+            XCTAssertNil(event.data?["data"])
+            expectation.fulfill()
+        }
+
+        // test
+        Optimize.updatePropositionsForSurfacePaths(["myView#htmlElement"], withXdm: nil)
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testUpdatePropositionsForSurfacePaths_validSurfaceWithXdmAndData() {
+        // setup
+        let expectation = XCTestExpectation(description: "updatePropositions should dispatch an event with expected data.")
+        expectation.assertForOverFulfill = true
+
+        let testEventData: [String: Any] = [
+            "requesttype": "updatepropositions",
+            "surfaces": [
+                [ "name": "myView#featureJson"
```

Comment:
How is this test working? `surfaces` should an array of strings right?

Reply:
Good catch! It's still working because testEventData is not being used. We are passing the surfaces in the API call and we are verifying the dispatched event data directly.